server/src/test/java/io/druid/server/coordinator/DruidCoordinatorTest.java

```diff
@@ -338,6 +338,10 @@ public void childEvent(
     Assert.assertEquals(1, segmentAvailability.size());
     Assert.assertEquals(0l, segmentAvailability.get(dataSource));
 
+    while (coordinator.getPendingSegments().get(dataSource).get() > 0) {
```

Comment:
It would be the cleaner solution but if this fix works and stops the sporadic failures, I'm fine with it